libs/ngx-charts-on-fhir/src/lib/timeline-range-selector/timeline-range-selector.component.ts

```diff
@@ -64,12 +64,14 @@ export class TimelineRangeSelectorComponent {
   }
 
   dateChange(event: MatDatepickerInputEvent<Date>, datePickerType: string) {
-    if (datePickerType === 'min') {
-      this.selectedDateRange = new DateRange(event.value, this.selectedDateRange.end);
-    } else {
-      this.selectedDateRange = new DateRange(this.selectedDateRange.start, event.value);
-    }
-    this.zoomChart();
+    setTimeout(() => {
+      if (datePickerType === 'min') {
+        this.selectedDateRange = new DateRange(event.value, this.selectedDateRange.end);
+      } else {
+        this.selectedDateRange = new DateRange(this.selectedDateRange.start, event.value);
+      }
+      this.zoomChart();
```

Comment:
If we delay setting `this.selectedDateRange`, the date values flicker briefly when you click Apply because it takes a couple change detection cycles for the state to stabilize. Can we put the `setTimeout` only around `zoomChart()`? So the local component state will change immediately, but it won't propagate the old end date into configService.